explorer-ui/src/components/ProfileView/ProfileDisplay/ProfileDisplay.jsx

```diff
@@ -28,59 +28,154 @@ export default function ProfileDisplay({userProfile}) {
+    setMatchesSortingComplete(true)
+    getCoordinates(rankedMatches)
+  }, [matches])
+    // TODO: Create a function that filters through the list of users to get the matches of this specific user
+    const getMatches = () => {
+      console.log("getting matches")
+      const profilesToAdd = new Set();
+      for (let i = 0; i <Object.keys(profiles).length; i++) {
+        console.log("user country: ", userProfile.country),
+          console.log("profile country: ", profiles[i].country)
+        if (
+          profiles[i].country === userProfile.country 
+          &&
+          profiles[i].travelMonth == userProfile.travelMonth &&
+          profiles[i].accomodation == userProfile.accomodation
+        ) {
+          console.log("it's a match!")
+          console.log("profiles[i]: ", profiles[i])
+          // profilesToAdd.push(profiles[i])
+          profilesToAdd.add(profiles[i])
+        }
+      }
+      setMatches([...matches, ...profilesToAdd])
+      console.log("profilesToAdd: ", profilesToAdd)
+      
+    };
+  
+      // Calculate the intersections between interests and attractions
+    function findIntersections(match){
+      console.log("find intersections")
+      if(match.preferenceInfo.interests && userProfile.preferenceInfo.interests){
+        console.log("match.preferenceInfo.interest: ", match.preferenceInfo.interests)
+        let interstIntersection = userProfile.preferenceInfo.interests.filter(x => match.preferenceInfo.interests.includes(x));
+        console.log("userProfile.preferenceInfo.interest: ", userProfile.preferenceInfo.interests)
+        let attractionIntersection = userProfile.preferenceInfo.attractions.filter(y =>  match.preferenceInfo.attractions.includes(y));
+        return {"interestIntersection": interstIntersection, "attractionIntersection": attractionIntersection, "match": match};
+      }
+      console.log("intersection: ", interstIntersection)
+      return {"interestIntersection": {}, "attractionIntersection": {}, "match": match};
+    }
+    
+    const rankMatches = (matches) =>{
+      console.log("rank matches")
+      console.log("matches are: ", matches)
+      if(matches){
+      // Create a state to keep track of the intersection data and iterate through each match to get that data
+      const matchesToRank = []
+      matches.forEach((match) => matchesToRank.push(findIntersections(match)))
+      matchesToRank.sort((a, b) => b.attractionIntersection.length - a.attractionIntersection.length)
+      setRankedMatches(matchesToRank)
+      console.log("ranked matches: ", rankedMatches)
+      console.log("matchesToRank: ", matchesToRank)
+      }
+    }
+
+    const getCoordinates = (rankedMatches) => {
+      if (rankedMatches) {
+        for (let i = 0; i < rankedMatches.length; i++) {
+          if(rankedMatches[i].match.address != null || rankedMatches[i].match.address !="" ){
+          
+          Geocode.fromAddress(rankedMatches[i].match.address).then(
```

Comment:
as doing a async request in a loop -> would create a helper fxn to call the API then in the for loop have it be blocking with async await

for(...) {
 await helperFxn();
}